alg/teca_2d_component_area.h

```diff
@@ -12,24 +12,38 @@ TECA_SHARED_OBJECT_FORWARD_DECL(teca_2d_component_area)
 
 /// an algorithm that computes the area of labeled regions
 /**
-Given a set of labels on a Cartesian mesh, the algorithm computes
-the area of each region. Regions are identified by assigning a
-unique integer value to each mesh point that belongs in the
-region. The component_variable property names the variable containing
-the region labels.
-
-if the region labels start at 0 and are contiguous then an
-optimization can be used. Set contiguous_component_ids property
-to enable the optimization.
-
-the input dataset is passed through and the results of the
-calculations are stored in the output dataset metadata in
-keys named:
-
-  number_of_components
-  component_ids
-  component_area
-
+Given a set of labels on a Cartesian mesh, the algorithm computes the area of
+each region. Regions are identified by assigning a unique integer value to each
+mesh point that belongs in the region. The component_variable property names
+the variable containing the region labels.
+
+if the region labels start at 0 and are contiguous then an optimization can be
+used. Set contiguous_component_ids property to enable the optimization. Note that
+TECA's connected component labeler assigns the background (i.e. cells not inside
+the segmentation) the label 0. One can identify the background region and area
+via this label. When processing data generated outside of TECA it might be
+necessary to supply the background label. Use -2 if there is no background.
```

Comment:
I don't immediately see any logic in the code handling the case of `this->background_id == 2`; all of the tests look for `this->background_id < 0`, in which case -1 wouldn't lead to different behavior form -2.   Is this comment correct, or am I misreading the code?

Reply:
yes, you are right. I'll address this now.